src/components/EntitlementsPanelForm/Fields/Fields.test.tsx

```diff
@@ -0,0 +1,162 @@
+import { screen, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Formik } from "formik";
+import { setupServer } from "msw/node";
+import { vi } from "vitest";
+
+import type { EntityEntitlement } from "api/api.schemas";
+import {
+  getGetResourcesMockHandler,
+  getGetResourcesResponseMock,
+} from "api/resources/resources.msw";
+import { renderComponent } from "test/utils";
+
+import { Label } from "../types";
+
+import Fields from "./Fields";
+
+const mockApiServer = setupServer(
+  getGetResourcesMockHandler(
+    getGetResourcesResponseMock({
+      data: [
+        {
+          id: "mock-id",
+          name: "editors",
+          entity: {
+            id: "mock-entity-id",
+            type: "mock-entity-name",
+          },
+        },
+      ],
+    }),
+  ),
+);
+const mockEntitlements = [
+  {
+    entity_type: "mock-entity-type-1",
+    entity_name: "",
+    entitlement_type: "mock-entitlement-type-1",
+  },
+  {
+    entity_type: "mock-entity-type-1",
+    entity_name: "",
+    entitlement_type: "mock-entitlement-type-2",
+  },
+  {
+    entity_type: "mock-entity-type-3",
+    entity_name: "",
+    entitlement_type: "mock-entitlement-type-3",
+  },
+];
+
+beforeAll(() => {
+  mockApiServer.listen();
+});
+
+afterEach(() => {
+  mockApiServer.resetHandlers();
+});
+
+afterAll(() => {
+  mockApiServer.close();
+});
+
+test("should show the correct fields initially", () => {
+  renderComponent(
+    <Formik<EntityEntitlement>
+      initialValues={{ entity_type: "", entity_name: "", entitlement_type: "" }}
+      onSubmit={vi.fn()}
+    >
+      <Fields entitlements={mockEntitlements} />
+    </Formik>,
+  );
+  const resourceTypeSelect = screen.getByRole("combobox", {
+    name: Label.ENTITY,
+  });
+  expect(resourceTypeSelect).toHaveValue("");
+  const resourceSelect = screen.getByRole("combobox", { name: Label.RESOURCE });
+  expect(resourceSelect).toHaveValue("");
+  expect(resourceSelect).toBeDisabled();
+  const entitlementTypeSelect = screen.getByRole("combobox", {
+    name: Label.ENTITLEMENT,
+  });
+  expect(entitlementTypeSelect).toHaveValue("");
+  expect(entitlementTypeSelect).toBeDisabled();
+});
+
+test("should display the correct values and populate all the fields", async () => {
```

Comment:
I think this would be better split up into three tests, so that we can be more explicit about testing that selecting option does the correct thing for the subsequent fields.

Reply:
I still think this is the case, this is a quite long and hard to read test.